app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb

```diff
@@ -19,14 +23,468 @@ def cluster_name_href(href)
+        cluster_service = connection.system_service.clusters_service.cluster_service(uuid_from_href(href))
+        networks = cluster_service.networks_service.list
+        networks.detect { |n| n.name == network_name }
+      end
+    end
+
+    def configure_vnic(args)
+      vm = args[:vm]
+      mac_addr = args[:mac_addr]
+      network = args[:network]
+      nic_name = args[:nic_name]
+      interface = args[:interface]
+      vnic = args[:vnic]
+      logger = args[:logger]
+
+      vm.with_provider_connection(:version => 4) do |connection|
+        uuid = uuid_from_href(vm.ems_ref)
+        profile_id = network_profile_id(connection, network.id)
+        nics_service = connection.system_service.vms_service.vm_service(uuid).nics_service
+        options = {
+          :name         => nic_name || vnic.name,
+          :interface    => interface || vnic.interface,
+          :mac          => mac_addr ? OvirtSDK4::Mac.new(:address => mac_addr) : vnic.mac,
+          :vnic_profile => profile_id ? { :id => profile_id } : vnic.vnic_profile
+        }
+        logger.info("with options: <#{options.inspect}>")
+        if vnic
+          nics_service.nic_service(vnic.id).update(options)
+        else
+          nics_service.add(OvirtSDK4::Nic.new(options))
+        end
+      end
+    end
+
+    def powered_off_in_provider?(vm)
+      vm.with_provider_object(:version => 4) { |vm_service| vm_service.get.status } == OvirtSDK4::VmStatus::DOWN
+    end
+
+    def powered_on_in_provider?(vm)
+      vm.with_provider_object(:version => 4) { |vm_service| vm_service.get.status } == OvirtSDK4::VmStatus::UP
+    end
+
+    def vm_boot_from_cdrom(operation, name)
+      operation.get_provider_destination.vm_service.start(
+        :vm => {
+          :os     => {
+            :boot => {
+              :devices => [OvirtSDK4::BootDevice::CDROM]
+            }
+          },
+          :cdroms => [
+            {
+              :id => name
+            }
+          ]
+        }
+      )
+    rescue OvirtSDK4::Error
+      raise OvirtServices::VmNotReadyToBoot
+    end
+
+    def vm_boot_from_network(operation)
+      operation.get_provider_destination.start(
+        :vm => {
+          :os => {
+            :boot => {
+              :devices => [
+                OvirtSDK4::BootDevice::NETWORK
+              ]
+            }
+          }
+        }
+      )
+    rescue OvirtSDK4::Error
+      raise OvirtServices::VmNotReadyToBoot
+    end
+
+    def get_template_proxy(template, connection)
+      TemplateProxyDecorator.new(
+        connection.system_service.templates_service.template_service(template.uid_ems),
+        connection,
+        self
+      )
+    end
+
+    def get_vm_proxy(vm, connection)
+      VmProxyDecorator.new(connection.system_service.vms_service.vm_service(vm.uid_ems), self)
+    end
+
+    def collect_disks_by_hrefs(disks)
+      vm_disks = []
+      ext_management_system.with_provider_connection(:version => 4) do |connection|
```

Comment:
Wondering if we should generate a constant for `:version => 4` to keep the process consistent.